vegafusion-python/tests/test_spark_e2e.py

```
@@ -0,0 +1,137 @@
+import pyarrow as pa
+from typing import Optional
+
+# TODO: cleanup this file
+
+# Root directory that contains the Vega-Lite spec fixtures used by this test suite
+SPEC_ROOT = (Path(__file__).parent / "specs").resolve()
+SALES_DATA_PATH = SPEC_ROOT / "sales_data_1kk.parquet"
+SALES_DATA_DF = pd.read_parquet(SALES_DATA_PATH)
+
+
+def _discover_spec_files(limit: Optional[int] = None) -> list[Path]:
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)